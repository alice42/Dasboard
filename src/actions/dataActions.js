export const DATA_REQUEST = 'DATA_REQUEST'
export const DATA_SUCCESS = 'DATA_SUCCESS'
export const DATA_ERROR = 'DATA_ERROR'

export const DATA_REQUESTA = 'DATA_REQUESTA'
export const DATA_SUCCESSA = 'DATA_SUCCESSA'
export const DATA_ERRORA = 'DATA_ERRORA'

export const dataRequest = () => {
  return {
    type: DATA_REQUEST
  }
}

export const dataRequestA = () => {
  return {
    type: DATA_REQUESTA
  }
}
