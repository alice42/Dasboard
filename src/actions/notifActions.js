export const NOTIF_REQUEST = 'NOTIF_REQUEST'
export const NOTIF_SUCCESS = 'NOTIF_SUCCESS'
export const NOTIF_ERROR = 'NOTIF_ERROR'

export const notifRequest = () => {
  return {
    type: NOTIF_REQUEST
  }
}
