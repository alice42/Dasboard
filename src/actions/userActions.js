export const AUTH_REQUEST = 'AUTH_REQUEST'
export const AUTH_SUCCESS = 'AUTH_SUCCESS'
export const AUTH_ERROR = 'AUTH_ERROR'

export const authRequest = data => {
  return {
    type: AUTH_REQUEST,
    data
  }
}
