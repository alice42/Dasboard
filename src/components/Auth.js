import React, { Component } from 'react'

class Auth extends Component {
  state = {
    identifiant: '',
    password: ''
  }
  handleSubmit = () => {
    const { identifiant, password } = this.state
    if (identifiant && password)
      this.props.userActions.authRequest({ identifiant, password })
  }
  handleChange = event => {
    const { target } = event
    const value = event.target.value
    const { name } = target
    this.setState({
      [name]: value
    })
  }
  render() {
    return (
      <div style={{ width: '100%', display: 'flex' }}>
        <div>
          identifiant
          <input
            type="text"
            name="identifiant"
            id="identifiant"
            placeholder="identifiant"
            value={this.state.identifiant}
            onChange={e => {
              this.handleChange(e)
            }}
          />
          password
          <input
            type="text"
            name="password"
            id="password"
            placeholder="password"
            value={this.state.password}
            onChange={e => {
              this.handleChange(e)
            }}
          />
          <button onClick={() => this.handleSubmit()}>login</button>
        </div>
      </div>
    )
  }
}

export default Auth
