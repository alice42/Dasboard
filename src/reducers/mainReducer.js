import { combineReducers } from 'redux'
import userReducer from './userReducer'
import notifReducer from './notifReducer'
import dataReducer from './dataReducer'

const mainReducer = combineReducers({
  user: userReducer,
  notif: notifReducer,
  data: dataReducer
})

export default mainReducer
