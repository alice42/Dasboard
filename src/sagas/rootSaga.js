import { all, fork } from 'redux-saga/effects'
import userSaga from './userSaga'
import notifSaga from './notifSaga'

function* rootSaga() {
  yield all([fork(userSaga), fork(notifSaga)])
}
export default rootSaga
