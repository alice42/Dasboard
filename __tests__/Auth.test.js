import "babel-polyfill"
import React from 'react'
import { render, fireEvent, waitFor, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { StateMock } from '@react-mock/state';

import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import createSagaMiddleware from 'redux-saga'
import mainReducer from '../src/reducers/mainReducer'
import rootSaga from '../src/sagas/rootSaga'

import * as userActions from '../src/actions/userActions'
import * as notifActions from '../src/actions/notifActions'
import * as dataActions from '../src/actions/dataActions'

import Auth from '../src/containers/Auth'

const sagaMiddleware = createSagaMiddleware()
const middleware = [sagaMiddleware]

composeWithDevTools(applyMiddleware(...middleware))
const store = createStore( 
  mainReducer,
  composeWithDevTools(applyMiddleware(...middleware))
)

describe('AUTH /', () => {

   it('should take a snapshot', () => {
       const props = store.getState()
       const { asFragment, getByText } = render(<Auth {...props} />)
       expect(asFragment(<Auth />)).toMatchSnapshot()
    })

   it('should render an error', async () => {
      const props = store.getState()
      props.user.error = 'ERROR MESSAGE'
      const { getByText } = render(
      <Auth {...props}
      dataActions={dataActions}
      notifActions={notifActions}
      userActions={userActions}
      />
      )
      expect(getByText('ERROR MESSAGE')).toBeInTheDocument()
   })

   it('events handlers',  async () => {
      const props = store.getState()
      const renderComponent = ({ identifiant, password }) =>
      render(
         <StateMock state={{ identifiant, password }}>
            <Auth {...props} 
            dataActions={dataActions}
            notifActions={notifActions}
            userActions={userActions}/>
         </StateMock>
      );

      const { getByText, getByPlaceholderText } = renderComponent({ identifiant: '', password: '' });

      fireEvent.click(getByText('Sign In'));
      await waitFor(() => {
         expect(getByPlaceholderText('identifiant').value).toEqual('')
         expect(getByPlaceholderText('password').value).toEqual('')
         })
      fireEvent.change(getByPlaceholderText('identifiant'), {
         target: { value: 'identifiantTest' }
      })
      fireEvent.change(getByPlaceholderText('password'), {
         target: { value: 'passwordTest' }
      })
      await waitFor(() => {
         expect(getByPlaceholderText('identifiant').value).toEqual('identifiantTest')
         expect(getByPlaceholderText('password').value).toEqual('passwordTest')
         }
      );
      fireEvent.click(getByText('Sign In'));
   })
})