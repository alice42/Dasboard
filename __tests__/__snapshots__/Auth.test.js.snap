// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTH / should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root css-161ifd5 MuiContainer-maxWidthXs"
    themespacing="64"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle css-1d3kzn2 MuiAvatar-colorDefault"
      color="[object Object]"
      themespacing="8"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <div
      class="css-18xyd39"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="identifiant"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="identifiant"
            name="identifiant"
            placeholder="identifiant"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-93 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-94"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="password"
            name="password"
            placeholder="password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-93 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-95"
            >
              <span>
                Password *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text css-14mvmu7 MuiButton-fullWidth"
        palette="[object Object]"
        tabindex="0"
        themespacing="24px 0px 16px"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Sign In
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
